test/e2e_tests/sync.go

```diff
@@ -15,7 +15,41 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const NamespaceManifestWorkPrefix = "mw-create-"
+const (
+	namespaceManifestWorkPrefix = "mw-create-"
+	envVarName                  = "E2E-TEST"
+	kindConfigMap               = "ConfigMap"
+	kindSecret                  = "Secret"
+)
+
+// verifySecretOrConfigMapCopy makes sure a Secret or a ConfigMap is eventually copied to the ManifestWork,
+// by creating a Capp with the needed configuration and getting the created ManifestWork resource.
+func verifySecretOrConfigMapCopy(baseCapp *cappv1alpha1.Capp, name, namespace, kind string) {
+	desiredCapp := utilst.CreateCapp(k8sClient, baseCapp)
+
+	assertionCapp := utilst.GetCapp(k8sClient, desiredCapp.Name, desiredCapp.Namespace)
+	Expect(assertionCapp.Name).ShouldNot(Equal(baseCapp.Name))
```

Comment:
I'm not sure this test should run on all the tests maybe check it once in a base test ... its checking the adapter function